GSA-TTS/FAC
# Replace missing major program in historical audits

Closes https://github.com/GSA-TTS/FAC/issues/3702

In this PR:
- Some historic audits are missing values for the `MAJORPROGRAM` field in the electaudits table, which (correctly) fails our validations.
- This fix sets `MAJORPROGRAM` to “Y” if `TYPEREPORT_MP` is provided, and “N” otherwise.
- Unit tests

Testing:
- "Y" case
  - Find (or create) an audit in the `elecaudits` table with a blank `MAJORPROGRAM` and has a `TYPEREPORT_MP` set.
  - Run `historic_data_migrator` on that audit and it should migrate without issue
  - The entry in the migrationinspectionrecord table should show "" for `MAJORPROGRAM` was replaced with "Y"
  - `is_major` in the dissemination_federalaward table should be "Y"
- "N" case 
  - Find (or create) an audit in the `elecaudits` table with a blank `MAJORPROGRAM` and does NOT have `TYPEREPORT_MP` set.
  - Run `historic_data_migrator` on that audit and it should migrate without issue
  - The entry in the migrationinspectionrecord table should show "" for `MAJORPROGRAM` was replaced with "N"
  - `is_major` in the dissemination_federalaward table should be "N"

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
